More IntelliJ fixes

magicus opened the pull request:
I now found out how to enable All Of The Checks! :-) I'm wading through false positives, but am actually finding quite a few gems.

magicus commented:
This sounds fantastic!!! I've been thinking about implementing some assistance to managing the bank, but never anything as ambitious as this, and I never got around to it. :(

The code look very clean and straightforward, too.

I have just a few, very minor, comments or questions:

I don't really understand where the quick access buttons are. Is it in the empty space next to the destination page numbers? Perhaps these needs to be made more visible. I also don't really get if this was an existing functionality? I've completely missed it if so.

What is the search highlight color? In the screenshot with "unravel", I can't see any item standing out..?

Finally, the "edit" pen symbol does not really feel like it belongs in the Wynncraft style. Perhaps replace it with a book and quill symbol? Or maybe the content team can help with producing a more fitting icon.

P0keDev commented:
> I don't really understand where the quick access buttons are. Is it in the empty space next to the destination page numbers? Perhaps these needs to be made more visible. I also don't really get if this was an existing functionality? I've completely missed it if so.

Yes, it's the empty space--it will also tell you in its tooltip where it will take you. It is a vanilla Wynn feature, but you're not the only person I know of who has missed it, so maybe adding a better indicator isn't a bad idea.


> Finally, the "edit" pen symbol does not really feel like it belongs in the Wynncraft style. Perhaps replace it with a book and quill symbol? Or maybe the content team can help with producing a more fitting icon.

Definitely a fair critique, I can try a book and quill. I believe there's already a symbol used in the quest book?

magicus commented:
To continue talking with myself :) I looked more closely at the code for the QA handling. I see my guess about the QA buttons were correct. Maybe you could populate the slots with an item that looks somewhat like a button? I'm somewhat out of ideas right now, but even something like a stone button would probably be an improvement over an empty space.

As a future follow-up, it would be great if the QA button could tie in with the page names and colors. Let's say you have a page 4 named PURPLE + "Build". Then the QA button could show a purple "B".

P0keDev commented:
> As a future follow-up, it would be great if the QA button could tie in with the page names and colors. Let's say you have a page 4 named PURPLE + "Build". Then the QA button could show a purple "B".

This is another good idea, I'll see what I can do.

HeyZeer0 commented:
> > As a future follow-up, it would be great if the QA button could tie in with the page names and colors. Let's say you have a page 4 named PURPLE + "Build". Then the QA button could show a purple "B".
> 
> This is another good idea, I'll see what I can do.

You could as well draw the text in the middle of the slot without shadow and with a gray accent. Would look good too ^^

magicus commented:
There is surprisingly no "write" symbol in the Quest Book what I can find (I thought there was one, too). There is however a "pen" symbol in the new character selection texture. It's still a pen and I honestly don't think it fits there either, but it's colored and I think it at least look a tad better.

HeyZeer0 commented:
> There is surprisingly no "write" symbol in the Quest Book what I can find (I thought there was one, too). There is however a "pen" symbol in the new character selection texture. It's still a pen and I honestly don't think it fits there either, but it's colored and I think it at least look a tad better.

Might removing the edit button is enough. You just have to make the title editable by clicking on

magicus commented:
@HeyZeer0 I see your point, but that severely lacks in discoverability. How will a user know that possibility exists? If given that option only, I prefer the current pen. ;-)

P0keDev commented:
I did some digging and found an unused texture in the Wynn pack that I think works for the quick access buttons: 
![image](https://user-images.githubusercontent.com/3767283/92198401-2af52480-ee29-11ea-9df1-78b05a575505.png)

I'm not entirely sure if my method of checking whether the texture is present (in the unlikely case that the user doesn't have the texture pack) is the cleanest method, but as far as I can tell the options are pretty limited. Will get to work on the other improvements soon.

P0keDev commented:
Now using the pencil from the class selector!
![image](https://user-images.githubusercontent.com/3767283/92216157-ccd93900-ee4a-11ea-8fbb-01f8476f5088.png)

magicus commented:
This looks really great! I'm happy with the fix as it is right now.

magicus commented:
@P0keDev I had an idea. Too late for this PR but certainly possible for a follow-up. Why don't you add the custom bank page name (if one exists) as a hover tooltip for the QA buttons? That should probably be quite easy to implement, and give you plenty of space to fill in the full name with colors and. everything, instead of trying to abbreviate it.

P0keDev commented:
> @P0keDev I had an idea. Too late for this PR but certainly possible for a follow-up. Why don't you add the custom bank page name (if one exists) as a hover tooltip for the QA buttons? That should probably be quite easy to implement, and give you plenty of space to fill in the full name with colors and. everything, instead of trying to abbreviate it.

I added this already, actually.

magicus commented:
@P0keDev Great minds think alike! :-) Fantastic work, thank you.

magicus committed:
Fix Scope is too broad

magicus committed:
Use less specific collection types for definition

magicus committed:
Move static method to where it is used

magicus committed:
Make utility classes final and add private constructor

magicus committed:
Various code quality fixes as suggested by IntelliJ

Examples include extracting large anonymous classes, fixing incorrect class/instance mixups, etc.

magicus committed:
Various code quality fixes

magicus committed:
Declaration access can be weaker

magicus committed:
More code improvements

Examples: make util classes final instead of abstract, don't use "equals" as overloaded name

magicus commented:
This PR is probably easiest to review per commit. The "major" commits are https://github.com/Wynntils/Artemis/pull/248/commits/5370f4fa56e98fb61d15e648cce45fe1f199c383 (which basically makes stuff private if it is possibly) and https://github.com/Wynntils/Artemis/pull/248/commits/b1c8fa7bf4270a300e9d361c75ee456c965968e5 (which makes utility classes final and adds a private constructor); these affect a lot of files with the same kind of changes. The rest are more of a mixed bag.

magicus committed:
Merge branch 'main' into more-intellij-fixes

magicus committed:
Merge branch 'main' into more-intellij-fixes

kristofbolyai commented on common/src/main/java/com/wynntils/commands/wynntils/WynntilsConfigCommand.java:
```diff
@@ -30,12 +30,14 @@
-public abstract class WynntilsConfigCommand {
-    public static final SuggestionProvider<CommandSourceStack> featureSuggestionProvider =
+public final class WynntilsConfigCommand {
+    private WynntilsConfigCommand() {}
```
Why?

kristofbolyai replied:
(talking about every empty constructor)

Incompleteusern replied:
I think the goal is to avoid unsubclassed abstract classes, which is done with the final + private constructor.

kristofbolyai replied:
Yeah but this is kinda annoying and messy?

magicus replied:
As @Incompleteusern says, it is to disallow instantiation. If you think it is too messy, I can remove it.

magicus replied:
(I agree that it is a bit messy, but ut also makes it clear it is a utility class. I'm on the fence for that one)

kristofbolyai replied:
This one kinda bothers me? What about @P0keDev @Incompleteusern @HighCrit ?

magicus replied:
If it bothers you, I'll remove it.

HighCrit replied:
In the past a decision was made to not have empty constructors on utility classes. When I proposed an empty constructor or one that throws an exception or both.

HighCrit replied:
I'm in favour of preventing instantiation on utility classes. But I won't lose a night's sleep over it. Wynntils isn't really used as a dependency anywhere and we're all capable enough to know or notice it in PRs.

magicus replied:
I just removed them, so, hey... :) I think they have some value, but as HighCrit says it also unlikely that we will make any mistakes regarding this, and they do add a bit of clutter. So let's avoid the clutter, then.

Incompleteusern committed:
Merge branch 'main' into more-intellij-fixes

Incompleteusern committed:
chore: spotless formatting

magicus committed:
Narrow down exception types in catch clause

(And a few other previously missed fixes)

magicus committed:
Add the inspection profile I've used

magicus committed:
Remove private no-args constructors from utility classes

kristofbolyai reviewed (approved):
Seems fine. Hope these changes won't cause big issues later on (although they look fine).

kristofbolyai committed:
Merge branch 'main' into more-intellij-fixes

kristofbolyai commented:
@magicus are we sure we need the inspection profile though? Isn't it just an "enable every setting" file?

magicus commented:
@kristofbolyai The inspection profile isn't strictly needed. It's just a shared profile that any of you can use when doing code analysis from IntelliJ. It contains the combination of enabled checks that I found useful. (I'm not saying all checks are perfect, or free from false positives, but they tend to catch real issues.) Stupid and/or irrelevant checks have been turned off.

I'd really like to see a way to go between the Qodona config and this inspection profile.
